cmake/toolchain/linux_x86_64_hpc.cmake

```diff
@@ -0,0 +1,7 @@
+set(BUILD_SHARED_LIBS OFF)
```

Comment:
Can this be called "linux_x86_64_static.cmake"?  This PR doesn't look like it has anything in it that's specific to HPC, rather just adding static support.